packages/grafana-ui/src/components/Logs/LogRowMessage.tsx

```diff
@@ -52,6 +54,44 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => {
   };
 });
 
+const restructureLog = (line: string, prettifyLogMessage: boolean): string => {
+  if (prettifyLogMessage) {
+    try {
+      const parser = getParser(line) as LogsParser;
+      const fields = parser.getFields(line);
+      return fields.join('\n');
+    } catch (error) {
+      return line;
+    }
+  }
+  return line;
+};
+
+function renderLogMessage(
+  needsHighlighter: boolean | '' | undefined,
```

Comment:
This is not related to your PR nit pick - just fyi - in `needsHighlighter` we could do this (which would make typing easier.
```
Boolean(
      highlights && highlights?.length > 0 && highlights[0] && highlights[0].length > 0 && entry.length < MAX_CHARACTERS
    )
```